src/Utils/InstrumentSchema.php

```diff
@@ -142,15 +142,46 @@ protected static function wrap_fields( $fields, $type_name ) {
 						do_action( 'graphql_before_resolve_field', $source, $args, $context, $info, $field_resolver, $type_name, $field_key, $field );
 
 						/**
-						 * If the current field doesn't have a resolve function, use the defaultFieldResolver,
-						 * otherwise use the $field_resolver
+						 * Create unique custom "nil" value which is different from the build-in PHP null, false etc.
+						 * When this custom "nil" is returned we can know that the filter did not try to preresolve
+						 * the field because it does not equal with anything but itself.
 						 */
-						if ( null === $field_resolver || ! is_callable( $field_resolver ) ) {
-							$result = Executor::defaultFieldResolver( $source, $args, $context, $info );
-						} else {
-							$result = call_user_func( $field_resolver, $source, $args, $context, $info );
+						$nil = new \stdClass();
+
+						/**
+						 * When this filter return anything other than the $nil it will be used as the resolved value
+						 * and the execution of the actual resolved is skipped. This filter can be used to implement
+						 * field level caches or for efficiently hiding data by returning null.
+						 *
+						 * @param mixed           $nil            Unique nil value
+						 * @param mixed           $source         The source passed down the Resolve Tree
+						 * @param array           $args           The args for the field
+						 * @param AppContext      $context        The AppContext passed down the ResolveTree
+						 * @param ResolveInfo     $info           The ResolveInfo passed down the ResolveTree
+						 * @param string          $type_name      The name of the type the fields belong to
+						 * @param string          $field_key      The name of the field
+						 * @param FieldDefinition $field          The Field Definition for the resolving field
+						 * @param mixed           $field_resolver The default field resolver
+						 */
+						$result = apply_filters( 'pre_graphql_resolve_field', $nil, $source, $args, $context, $info, $type_name, $field_key, $field, $field_resolver );
```

Comment:
let's do `graphql_pre_resolve_field`.

I've been trying to keep all the filters prefixed with `graphql_`